tokio-util/src/io/sink_writer.rs

```diff
@@ -0,0 +1,117 @@
+use futures_sink::Sink;
+
+use pin_project_lite::pin_project;
+use std::io;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use tokio::io::AsyncWrite;
+
+pin_project! {
+    /// Convert a [`Sink`] of byte chunks into an [`AsyncWrite`].
+    ///
+    /// Whenever you write to this [`SinkWriter`], the supplied bytes are
+    /// forwarded to the inner [`Sink`]. When `shutdown` is called on this
+    /// [`SinkWriter`], the inner sink is closed.
+    ///
+    /// This adapter implements [`AsyncWrite`] for `Sink<&[u8]>`. If you want to
+    /// implement `Sink<_>` for [`AsyncWrite`], see the [`codec`] module; if you need to implement
+    /// [`AsyncWrite`] for `Sink<Bytes>`, see [`CopyToBytes`].
```

Comment:
```suggestion
    /// This adapter implements [`AsyncWrite`] for `Sink<&[u8]>`. Because of the
    /// lifetime, this trait is relatively rarely implemented. The main ways to
    /// get a `Sink<&[u8]>` that you can use with this type are:
    ///
    ///  * With the codec module by implementing the `Encoder<&[u8]>` trait.
    ///  * By wrapping a `Sink<Bytes>` in a [`CopyToBytes`].
    ///  * Manually implementing `Sink<&[u8]>` directly.
    ///
    /// The opposite conversion of implementing `Sink<_>` for an [`AsyncWrite`]
    /// is done using the [`codec`] module.
```